contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -30,17 +30,61 @@ def register_options(cls, register):
     register(
       '--install-optional', type=bool, default=False, fingerprint=True,
       help='If enabled, install optional dependencies.')
+    register(
+      '--production-only', type=bool, default=False, fingerprint=True,
```

Comment:
There is an `--install-optional` flag, which implies that maybe this should be `--install-production-only`? Alternatively, could rename that flag to `--optional` and mark the old one deprecated? Example: https://github.com/pantsbuild/pants/blob/master/testprojects/src/python/plugins/dummy_options/tasks/dummy_options.py#L20-L21